src/mono/mono/mini/mini-runtime.c

```diff
@@ -5395,3 +5401,33 @@ mini_alloc_jinfo (MonoJitMemoryManager *jit_mm, int size)
 		return (MonoJitInfo*)mono_mem_manager_alloc0 (jit_mm->mem_manager, size);
 	}
 }
+
+void
+mono_set_runtime_init_callback (MonoRuntimeInitCallback callback)
+{
+	runtime_init_callback = callback;
+}
+
+/*
+* Default implementation invoking runtime init callback in native-to-managed wrapper.
+*/
+void
+mono_invoke_runtime_init_callback (void)
+{
+	if (G_UNLIKELY (runtime_init_callback)) {
```

Comment:
this needs to be an atomic read acquire, I think

Reply:
Yes we could do acquire/release semantics on runtime_init_callback, the atomic operations around this would prevent reordering, but putting in acquire/release semantics will make it more clear.